### ISSUE TITLE
NullableToStringCall: ignore by lambda receiving function name

### ISSUE THREAD

**Issue by vladimirfx** [detekt/detekt#6608]
## Expected Behavior
Configure ignorance of the `NullableToStringCall` rule by a function that receives lambda with issues.

## Current Behavior
We can't ignore the huge volume of cases where lambda can contain `null` in strings such as logging statements.

## Context

After the fix for NullableToStringCall in 1.23.3 (#6352) we see thousands of issues with over 95% for valid logging cases such as:

```kotlin
log.debug { "Committing offsets for previously delivered records: $deliveredOffsets." }
```
In this case absolutely okay to see `null` in logs. Moreover, it is good to see exactly `null` and not some synthetic value. So we need the ability to exclude `log.debug{}`, `log.info{}`, `log.warn{}` etc statements. It is the same thing as ignorance in `toString` - it is for technical not for the end user.

This is possible at all? Or may some simplified flag for disabling in all lambdas (pre #6352 behavior).

**Comment by atulgpt** [comment]
@BraisGabin @cortinico is there a workaround here? My project also uses a custom logger like `BBLogUtils.log{ "Some log with $id" }` and now those places are getting flagged as a violation.

**Comment by BraisGabin** [comment]
I think that we should have a suppressor for these. Suppress the issue is inside the parameters of certain functions.

**Comment by atulgpt** [comment]
Hi @BraisGabin is this possible as of now with current list of suppressors of it is something that we will be doing this with this bug?

**Comment by BraisGabin** [comment]
That's a suppressor that we should implement to solve this.

**Comment by BraisGabin** [comment]
Closing in favor of #5510. This is just another insurance of the need of that suppressor

**Comment by atulgpt** [comment]
HI, @BraisGabin will this require a new suppressor or suppression like below will work?

```kotlin
fun log(lazyBlock: @Suppress("NullableToStringCall") () -> Unit) // we have suppressed at the lambda declaration only
```

**Comment by BraisGabin** [comment]
That would not work
